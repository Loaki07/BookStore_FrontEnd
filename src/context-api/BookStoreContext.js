import React, { useState, createContext } from 'react';

export const BookStoreContext = createContext();

export const BookStoreProvider = (props) => {
  const [state, setState] = useState({
    loginUserEmailId: '',
    loginUserPassword: '',
    toggleLogInSignUp: false,
    message: 'baseState',
  });

  return (
    <div>
      <BookStoreContext.Provider value={[state, setState]}>
        {props.children}
      </BookStoreContext.Provider>
    </div>
  );
};
