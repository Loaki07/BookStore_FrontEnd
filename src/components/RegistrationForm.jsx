import React, { useContext } from 'react';
import { BookStoreContext } from '../context-api/BookStoreContext'
import { Formik, Form as FormikForm, Field, ErrorMessage } from 'formik'
import * as Yup from 'yup'
import FormErrorMessage from './FormErrorMessage.jsx';
import UserApis from "../services/UserApis"
import { Form, Button } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';
import {toast} from 'react-toastify';
import 'react-toastify/dist/ReactToastify.min.css';
toast.configure()

const { registerNewUser } = new UserApis();

const initialValues = {
  fullName: "",
  emailId: "",
  password: "",
  mobileNumber: "",
}

const validationSchema = Yup.object().shape({
  fullName: Yup.string()
                .min(3, 'Minimum 3 characters required!')
                .matches(/^[A-Za-z]+$/, "Cannot contain numbers!")
                .required('Required!'),
  emailId: Yup.string()
              .email("Invalid email format")
              .required("Required!"),
  password: Yup.string()
               .min(6, 'Minimum 6 characters required!')
               .required('Required!'),
  mobileNumber: Yup.string()
                   .min(10, 'Minimum 10 characters required!')
                   .required('Required!'),
})

const RegistrationForm = (props) => {
const [state, setState] = useContext(BookStoreContext);

  const handleFormChange = (e) =>{
    setState({ ...state, [e.target.name]: e.target.value})
  }

  const handleFormSubmit = async () => {
    try {
      console.log("Form Register", state); 
      const registerUserObject = {
        fullName: state.fullName,
        emailId: state.emailId,
        password: state.password,
        mobileNumber: state.mobileNumber
      }
      const result = await registerNewUser(registerUserObject);
      if (result.status === 200) {
        toast.success('Registration Successfull !', {position: toast.POSITION.TOP_CENTER});
        // props.history.push('/');
        console.log(result);
      }
    } catch (error) {
      toast.error(error.message, {position: toast.POSITION.TOP_CENTER});
    }
  } 

  return (
    <Formik 
      initialValues={initialValues}
      validationSchema={validationSchema}
      onSubmit={handleFormSubmit}
      >
        {({ handleSubmit, handleChange, values }) => (
          <FormikForm>
            <Form.Group>
              <Form.Label>Full Name</Form.Label>
              <Field 
                  type="text" 
                  placeholder="Full Name" 
                  className="form-control"
                  name="fullName" 
                  value={values.fullName} 
                  onInput={handleChange}
                  onChange={handleFormChange} 
                  autoComplete="off" 
                ></Field>
                <ErrorMessage 
                  name="fullName" 
                  component={FormErrorMessage} 
                ></ErrorMessage>
            </Form.Group>
            <Form.Group>
              <Form.Label>Email Id</Form.Label>
                <Field 
                      type="email" 
                      placeholder="Email Id" 
                      className="form-control"
                      name="emailId" 
                      value={values.emailId} 
                      onInput={handleChange}
                      onChange={handleFormChange} 
                      autoComplete="off" 
                    ></Field>
                    <ErrorMessage 
                      name="emailId" 
                      component={FormErrorMessage} 
                    ></ErrorMessage>
            </Form.Group>
            <Form.Group>
            <Form.Label>Password</Form.Label>
              <Field 
                    type="password" 
                    placeholder="Password" 
                    className="form-control"
                    name="password" 
                    value={values.password} 
                    onInput={handleChange}
                    onChange={handleFormChange} 
                    autoComplete="off" 
                  ></Field>
                  <ErrorMessage 
                    name="password" 
                    component={FormErrorMessage} 
                  ></ErrorMessage>
            </Form.Group>
            <Form.Group>
            <Form.Label>Mobile Number</Form.Label>
              <Field 
                    type="text" 
                    placeholder="Mobile Number" 
                    className="form-control"
                    name="mobileNumber" 
                    value={values.mobileNumber} 
                    onInput={handleChange}
                    onChange={handleFormChange} 
                    autoComplete="off" 
                  ></Field>
                  <ErrorMessage 
                    name="mobileNumber" 
                    component={FormErrorMessage} 
                ></ErrorMessage>
            </Form.Group>
            <Form.Group>
              <Button 
                variant='danger'
                type='submit'
                onSubmit={handleSubmit}
                block
              >Signup</Button>
            </Form.Group>
          </FormikForm>
        )}
    </Formik>
  )
}

export default RegistrationForm
