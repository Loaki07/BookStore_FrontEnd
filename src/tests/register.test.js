import React, { useContext } from 'react';
import { configure, mount, shallow } from 'enzyme';
import { render, screen } from '@testing-library/react';
import Adapter from 'enzyme-adapter-react-16';
import { BookStoreProvider, BookStoreContext } from '../context-api/BookStoreContext';
import RegistrationForm from '../components/RegistrationForm.jsx';

configure({ adapter: new Adapter() });

describe('Login', () => {
  const wrapper = mount(
    <BookStoreProvider>
      <RegistrationForm />
    </BookStoreProvider>
  );
  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should have an fullName field', () => {
    expect(wrapper.find('Field[name="fullName"]').length).toEqual(1);
  });

  it('should have an emailId field', () => {
    expect(wrapper.find('Field[name="emailId"]').length).toEqual(1);
  });

  it('should have an password field', () => {
    expect(wrapper.find('Field[name="password"]').length).toEqual(1);
  });
  
  it('should have an mobileNumber field', () => {
    expect(wrapper.find('Field[name="mobileNumber"]').length).toEqual(1);
  });
});