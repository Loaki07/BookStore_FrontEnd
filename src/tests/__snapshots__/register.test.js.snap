// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot 1`] = `"<div><form action=\\"#\\"><div class=\\"form-group\\"><label class=\\"form-label\\">Full Name</label><input name=\\"fullName\\" type=\\"text\\" placeholder=\\"Full Name\\" class=\\"form-control\\" autocomplete=\\"off\\" value=\\"\\"></div><div class=\\"form-group\\"><label class=\\"form-label\\">Email Id</label><input name=\\"emailId\\" type=\\"email\\" placeholder=\\"Email Id\\" class=\\"form-control\\" autocomplete=\\"off\\" value=\\"\\"></div><div class=\\"form-group\\"><label class=\\"form-label\\">Password</label><input name=\\"password\\" type=\\"password\\" placeholder=\\"Password\\" class=\\"form-control\\" autocomplete=\\"off\\" value=\\"\\"></div><div class=\\"form-group\\"><label class=\\"form-label\\">Mobile Number</label><input name=\\"mobileNumber\\" type=\\"text\\" placeholder=\\"Mobile Number\\" class=\\"form-control\\" autocomplete=\\"off\\" value=\\"\\"></div><div class=\\"form-group\\"><button type=\\"submit\\" class=\\"btn btn-danger btn-block\\">Signup</button></div></form></div>"`;
